Documentation/ApiOverview/ContentElements/Index.rst

```diff
@@ -110,21 +110,86 @@ your projects.
 possible without writing PHP code and can therefore also be done by
 TYPO3 integrators.
 
+..  _plugins:
 
+Plugins in TYPO3
+----------------
+
+A **plugin** in TYPO3 is a more complex module, typically providing dynamic
+or interactive functionality. Plugins are usually provided by extensions
+that introduce new features to the website.
+
+The data to be displayed is usually supplied by a special PHP class
+called a "controller". Depending on the technology used in the controller
+the plugin can be an Extbase plugin or a plain plugin.
 
-..  _plugins:
 ..  _plugins-extbase:
-..  todo: Introduce Extbase plugins
 
-What are plugins?
------------------
+Extbase plugins
+~~~~~~~~~~~~~~~
+
+For usage in the TYPO3 backend Extbase plugins are registered with utility
+functions of class :php:`\TYPO3\CMS\Extbase\Utility\ExtensionUtility` (not to
+be confused with :php:`\TYPO3\CMS\Core\Utility\ExtensionManagementUtility`).
+
+An Extbase plugin is configured for the frontend with
+:php:`ExtensionUtility::configurePlugin()` in file
+:file:`EXT:my_extension/ext_localconf.php`:
+
+..  literalinclude:: _Plugins/_ext_localconf_extbase_plugin.php
+    :caption: EXT:my_extension/ext_localconf.php
+
+By using `ExtensionUtility::PLUGIN_TYPE_PLUGIN` as fifth parameter is is also
+possible to add the plugin as a list type. See :ref:`plugins-list_type`.
+
+Method :php:`ExtensionUtility::configurePlugin()` also takes care of registering
+the plugin for frontend output in TypoScript using an object of type
+:ref:`EXTBASEPLUGIN <t3tsref:cobj-extbaseplugin>`.
+
+If it is desired that editors can insert the Extbase plugin like a content
+element into the page it also needs to be registered with
+:php:`ExtensionUtility::registerPlugin()` in the TCA Overrides, for example file
+:file:`EXT:my_extension/Configuration/TCA/Overrides/tt_content.php`:
+
+..  literalinclude:: _Plugins/_tt_content_extbase_plugin.php
+    :caption: EXT:my_extension/Configuration/TCA/Overrides/tt_content.php
+
+For a detailed discussion of Extbase plugins including examples for controllers
+see chapter :ref:`extbase`.
+
+..  _plugins-non-extbase:
+
+Plugins without Extbase
+~~~~~~~~~~~~~~~~~~~~~~~
+
+It is possible to create a plugin without using Extbase by creating a plain PHP
+class as a controller.
+
+In this case you have to define the TypoScript configuration yourself. A
+:ref:`USER or USER_INT <t3tsref:cobj-user-int>` TypoScript object can be used
+to delegate the rendering to your controller:
+
+..  literalinclude:: _Plugins/plugin.typoscript
+    :caption: EXT:my_extension/Configuration/TypoScript/setup.typoscript
+
+To register such a plugin as content element you can use function
+:php:`ExtensionManagementUtility::addPlugin()` in the TCA overrides, for example
```

Comment:
Somewhere above you have written about the pretty cool fifth argument to mark plugin as list-type. Why not using this fifth argument to mark plugin as content element? PLUGIN_TYPE_CONTENT_ELEMENT

Reply:
Sorry. My fault. We are in Plugins without Extbase. Holy. Totally confused.

Reply:
Ack